core/state_test.go

```diff
@@ -526,6 +571,51 @@ func TestRevert(t *testing.T) {
 	})
 }
 
+func TestRevertSpecialContracts(t *testing.T) {
+	client, closeFn := feeder.NewTestClient(utils.MAINNET)
+	t.Cleanup(closeFn)
+
+	gw := adaptfeeder.New(client)
+
+	testDB := pebble.NewMemTest()
+	txn := testDB.NewTransaction(true)
+	t.Cleanup(func() {
+		require.NoError(t, txn.Discard())
+	})
+
+	state := core.NewState(txn)
+
+	su0, err := gw.StateUpdate(context.Background(), 0)
+	require.NoError(t, err)
+
+	require.NoError(t, state.Update(0, su0, nil))
+
+	su1, err := gw.StateUpdate(context.Background(), 1)
+	require.NoError(t, err)
+
+	// These value were taken from part of integration state update number 299762
+	// https://external.integration.starknet.io/feeder_gateway/get_state_update?blockNumber=299762
+	scKey := utils.HexToFelt(t, "0x492e8")
+	scValue := utils.HexToFelt(t, "0x10979c6b0b36b03be36739a21cc43a51076545ce6d3397f1b45c7e286474ad5")
+	scAddr := new(felt.Felt).SetUint64(1)
+
+	t.Run("revert state update with special contract", func(t *testing.T) {
+		// update state root
+		su1.NewRoot = utils.HexToFelt(t, "0x2829ac1aea81c890339e14422fe757d6831744031479cf33a9260d14282c341")
```

Comment:
Was this calculated by hashing `su0.NewRoot` with `0x1`?

Reply:
Yes, I calculated it by running the test with `0x1` and when it failed I copied the root it failed with and updated the `NewRoot` (assumption: our trie's root calculation is correct).

Reply:
No, on line 606 I update the storage diff of `su1` with `0x1` and without updating the root `Update` would fail, as expected. Thus I use the root `Update` fails with as `NewRoot`. I make the assumption that `Update` is correct and `Trie` correctly propagates values.